pkg/ccl/backupccl/restore_planning.go

```diff
@@ -599,6 +599,12 @@ func allocateDescriptorRewrites(
 
 			if _, ok := restoreDBNames[targetDB]; ok {
 				needsNewParentIDs[targetDB] = append(needsNewParentIDs[targetDB], table.ID)
+				// We need to rewrite the Public schema id from 29 to the ID of the actual
```

Comment:
General question about all the changes in this method. Do we need them considering we in `restore_job.go` we are:
- iterating over all descriptor rewrites
-  checking if they lie in a parentDB without a physical public desc
- checking if they lie in the old public schema ID 29
- setting ParentSchemaID to the synthesized public descriptor.

I'm not sure I understand the changes in the planning phase of restore. Can't we do everything in the restore_job `remapSchemas`, and change _almost_ nothing in restore_planning. As long as we have the `ParentSchemaID` for all schema objects set to the correct value during planning, which we should after https://github.com/cockroachdb/cockroach/pull/72959.